src/features/NavBar/Search.jsx

```diff
@@ -55,46 +55,47 @@ const Search = () => {
     <form
       onSubmit={handleSearchSubmit}
       autoFocus
-      className={`city_box--search search-box relative w-1/2 h-12 flex-row justify-evenly items-center ${
+      className={` city_box--search search-box  w-1/2 h-12 flex-row justify-evenly items-center ${
         searchFieldVisible ? 'flex' : 'hidden'
       }`}
     >
-      <input
-        onFocus={() => setIsInputFocused(true)}
-        onBlur={() => setIsInputFocused(false)}
-        list="search-history"
-        ref={inputRef}
-        placeholder="Enter a city"
-        className="search-box--input rounded active:border active:border-slate-600 active:outline-none h-10 w-2/3 p-1 placeholder:opacity-50 text-black"
-      />
-
-      <ul
-        className={`search-box--history history absolute top-12 left-8 w-2/3 transition-all duration-200 ${
-          isInputFocused && searchHistory.length > 0
-            ? 'h-auto opacity-100'
-            : 'h-0 opacity-0'
-        } bg-slate-50 border rounded p-2 z-50`}
-      >
-        {searchHistory.map((place, i) => (
-          <li
-            onClick={() => handleHistoryItem(place)}
-            key={i}
-            className="history--item cursor-pointer flex flex-row justify-between opacity-[inherit] hover:bg-slate-100"
-          >
-            <p className="text-black">{place}</p>
-            <button
-              type="button"
-              onClick={e => {
-                e.stopPropagation();
-                handleDeleteHistoryItem(i);
-              }}
-              className=""
+      <div className="w-2/3 relative">
+        <input
+          onFocus={() => setIsInputFocused(true)}
+          onBlur={() => setIsInputFocused(false)}
+          list="search-history"
+          ref={inputRef}
+          placeholder="Enter a city"
+          className="search-box--input rounded active:border active:border-slate-600 active:outline-none h-10 w-full p-1 placeholder:opacity-50 text-black"
+        />
+        <ul
+          className={`absolute search-box--history history w-full transition-all duration-200 ${
+            isInputFocused && searchHistory.length > 0
+              ? 'h-auto opacity-100'
+              : 'h-0 opacity-0'
+          } bg-slate-50 border rounded p-2 z-50`}
+        >
+          {searchHistory.map((place, i) => (
+            <li
+              onClick={() => handleHistoryItem(place)}
+              key={i}
+              className="history--item cursor-pointer flex flex-row justify-between opacity-[inherit] hover:bg-slate-100"
             >
-              <CiCircleRemove className="delete-icon" />
-            </button>
-          </li>
-        ))}
-      </ul>
+              <p className="text-black">{place}</p>
+              <button
+                type="button"
+                onClick={e => {
+                  e.stopPropagation();
+                  handleDeleteHistoryItem(i);
```

Comment:
why do you call e.stopPropagation ?

Reply:
separated delete button and Location selection button